frontend/class-handle-404.php

```diff
@@ -62,6 +67,28 @@ private function is_feed_404( $handled ) {
 		return true;
 	}
 
+	/**
+	 * Determine whether this is the main feed.
+	 *
+	 * @global WP       $wp
+	 * @global WP_Query $wp_query
+	 *
+	 * @return bool Whether or not the request is the main feed.
+	 */
+	private function is_main_feed() {
+		global $wp, $wp_query;
+
+		if ( $wp_query->is_archive() || $wp_query->is_singular() || $wp_query->is_search() ) {
+			return false;
+		}
+
+		if ( array_diff( array_keys( $wp->query_vars ), array( 'feed' ) ) && ! $wp_query->is_comment_feed() ) {
```

Comment:
Please replace `array_diff( array_keys( $wp->query_vars ), array( 'feed' ) )` by `array_key_exists( ‘feed’, $wp->query_vars );` to increase readability.

Reply:
@IreneStr It should match only when array contains 'feed' (without other query keys). I could use:
```php
if ( array_keys( $wp->query_vars ) !== array( 'feed' ) && && !$wp_query->is_comment_feed() ) {
```
I can't find shorter form. Is it acceptable?

Reply:
Ah, in that case, your suggestion seems fine to me :)

Reply:
I just found more readable code. I'll think over again and you can send feedback:
```php
    $feed_query_vars  = array(
        'feed'         => '',
        'withcomments' => '',
    );
    $other_query_vars = array_diff_key( $wp->query_vars, $feed_query_vars ); 

    // Return true if there aren't other query variables.
    return empty( $other_query_vars );
```
Anyway, I'll try to check how it works in combination with other plugins and I'll try to create skeleton for php unit tests until we are thinking/checking.

Reply:
@stodorovic I think `if ( array_keys( $wp->query_vars ) !== array( 'feed' ) && ! $wp_query->is_comment_feed() )` is more clear, because using that syntax, the two things you want to check for (no other query_var other than 'feed', and it's no 'comment_feed') remain two separate things.